### DISCUSSION TITLE
Make a "macros" property Spatie\RouteAttributes\Attributes\Route so we can execute custom code for a route

### DISCUSSION THREAD

**Discussion by erickcomp** [spatie/laravel-route-attributes#122] [Ideas]
Many of the routes packages create macros around the ```\Illuminate\Routing\Route``` class. And even for implementing something for a specific project To improve compatibility with such situations, it would be good to have a "macros" param on the ```Spatie\RouteAttributes\Attributes\Route::__construct``` method (which, of course, implies having a $macros property.

I though about the format (but I have to give it a little more thinking.):
```php
#[Get( '/', name: 'home', middleware: [MyFancyMiddleware::class] macros: [RouteMacro('breadcrumb', 'Home', 'home')])]
```

Then, in the RouteRegistrar class (https://github.com/spatie/laravel-route-attributes/blob/1df42b60a6c282f504309d3d98f68474043afabd/src/RouteRegistrar.php#L172), we could call those macros in the route object. This way we could execute a lot of custom code in the Route registering process.

If you see any problems with this, please let me know. If that's too priority for you guys, maybe I can implement this some point in the future and submit a PR.

**Comment by freekmurze** [comment]
Feel free to submit a PR. If the code is simple, well tested and documented, I'd consider pulling it in 👍
